### ISSUE TITLE
Testing with Jest on components with RTK-query doesn't catch errors

### ISSUE THREAD

**Issue by ymoon715** [reduxjs/redux-toolkit#2415]
I'm currently writing tests in jest and testing-library, while using msw to mock my results as well as errors.
On a browser, it catches errors fine, and returns the results fine as well.

 https://github.com/ymoon715/rtk-query-msw

When mocking endpoints with msw on jest, it properly returns the data I'm trying to mock.
However, my lazy query's result of error, isError, isSuccess, are all false, while isUnitialized is true. When I'm trying to see if it's properly catching the error, it fails to catch the error.

I have a component here where if one of the endpoint returns an error, I display a snackbar with an error message. However, It never reaches the catch exception.

It seems like intercepting results through node doesn't work well with rtk-query.
Anyone running into the same issue? I've tried just about everything, including chaining .then and .catch after unwrap(). 

To Summarize the issue:
* Stubbing success response for lazy query returns proper response, but **isSuccess is false, and isUninitialized is false.** It triggers transformResponse properly.
* Stubbing error response for lazy query doesn't return proper error response, doesnt trigger catch exception, instead returns undefined as if the response was successful but it returned undefined, but **isError and isSuccess is false, and isUnitialized is true**. It does NOT trigger transformResponse.
* When testing it in the browser, it works properly and catches error properly as well.

**Comment by ymoon715** [comment]
Anyone try writing component tests for rtk-query and experience the same issue when testing for error cases? If it worked for you, let me know, and I'll go ahead and close this issue. I just haven't seen anything online regarding this.

**Comment by markerikson** [comment]
Can you put together a CodeSandbox or repo that reproduces the issue?  You could also use Replay ( https://replay.io/record-bugs ) to record this and share the recording link publicly with us.

**Comment by yukihirop** [comment]
I haven't tried it, but isn't the rtk query cache suspicious?

```ts
// .jest/setupAfter.ts
afterEach(()=>{
  store.dispatch(api.util.resetApiState());
})
```

https://medium.com/@t3j3ndra/how-to-clear-rtk-query-cache-in-tests-between-requests-when-using-mock-service-worker-and-jest-cb36ea4ffb4b
